gobblin-cluster/src/main/java/gobblin/cluster/GobblinClusterManager.java

```diff
@@ -165,6 +167,26 @@ public GobblinClusterManager(String clusterName, String applicationId, Config co
     this.appWorkDir = appWorkDirOptional.isPresent() ? appWorkDirOptional.get() :
         GobblinClusterUtils.getAppWorkDirPath(this.fs, clusterName, applicationId);
 
+    // create a job catalog for keeping track of received jobs if a job config path is specified
+    if (config.hasPath(GobblinClusterConfigurationKeys.GOBBLIN_CLUSTER_PREFIX
+        + ConfigurationKeys.JOB_CONFIG_FILE_GENERAL_PATH_KEY)) {
+      String jobCatalogClassName = ConfigUtils.getString(config, GobblinClusterConfigurationKeys.JOB_CATALOG_KEY,
+          GobblinClusterConfigurationKeys.DEFAULT_JOB_CATALOG);
+
+      this.jobCatalog = (MutableJobCatalog)GobblinConstructorUtils.invokeFirstConstructor(Class.forName(
+          jobCatalogClassName),
+            ImmutableList.<Object>of(config.getConfig(StringUtils.removeEnd(
+                GobblinClusterConfigurationKeys.GOBBLIN_CLUSTER_PREFIX, "."))));
+
+      // other services such as the job configuration manager have a dependency on the job catalog, so it has be be
+      // started first
+      if (this.jobCatalog instanceof Service) {
```

Comment:
should this be started as the first service in the startUp() instead, or is it mandatory to start it early on at construct time?